### PR TITLE
Move null casting tests to main model tests file

### PR DESCRIPTION
**Pull request by JosephSilber**
No need for that big separate test file, which also happens to duplicate all the non-null casting tests in the eloquent model test file.

### CODE REVIEW

File: tests/Database/DatabaseEloquentModelTest.php [hunk 1]
```diff
@@ -1152,6 +1152,40 @@ public function testModelAttributesAreCastedWhenPresentInCastsArray()
 	}
 
 
+	public function testModelAttributeCastingPreservesNull()
```
Comment (GrahamCampbell) [inline tests/Database/DatabaseEloquentModelTest.php]: Grrr. I really hate laravel's tests like this. Way to much in a single test. We should be using data providers or something.

Comment (JosephSilber) [inline tests/Database/DatabaseEloquentModelTest.php]: Maybe, but this is basically what's throughout the whole codebase.

Unless we're doing a sweeping change everywhere else, there's no need for this to be so different.

Comment (GrahamCampbell) [inline tests/Database/DatabaseEloquentModelTest.php]: Yeh, I know. I was complaining about the whole code base, lol.

Comment (JosephSilber) [inline tests/Database/DatabaseEloquentModelTest.php]: I happen to quite like it. In some cases you can get quite carried away, but in this case I think having it all in one method makes perfect sense.

All code in this method runs a test for one single line of code.

Comment (GrahamCampbell) [inline tests/Database/DatabaseEloquentModelTest.php]: That's not how tests are meant to be though. You're not meant to squash the test suite to try to match the implementation. You're writing tests that check different things. They shouldn't be organised in a way that nessessaraly reflects the implementation.

Comment (JosephSilber) [inline tests/Database/DatabaseEloquentModelTest.php]: The way I see it, I'm testing one single thing: **no casting should be done for `null`**. Period.

Comment (GrahamCampbell) [inline tests/Database/DatabaseEloquentModelTest.php]: Yep. So data providers are perfect. We'd still have one test method.

**Pull request merged**